examples/example2.json.txt

```diff
@@ -49,28 +49,37 @@ h'00112233445566778899aabbccddeeff0123456789abcdeffedcba9876543210'
                 / algorithm-id / -16 / "sha256" /,
                 / digest-bytes /
 h'3ee96dc79641970ae46b929ccf0b72ba9536dd846020dbdc9f949d84ea0e18d2'
+            ],
+            / text / 13:[
```

Comment:
I don't see why text has beed moved before the executable metadata. This does not appear to me to be an improvement.

Reply:
This is similar to https://github.com/suit-wg/manifest-spec/pull/48#discussion_r857485975

According to the CDDL, text appears before the unseverable members.

I realize that, according to canonical CBOR and map ordering, the text "jumps" to the end of the manifest because of its higher number, and I won't say definitely that it doesn't belong there, but it's sort of confusing that with canonical CBOR we are "locked" to one ordering, but it's a different order than appears in the CDDL. And if someone adds extensions, text will no longer be last anyway. And, with everything being wrapped in `bstr`s, skipping the text is quite trivial.

I'm bringing this up because, when generating code with zcbor (previously cddl-gen), there is no logic for making `text` "jump" to the end in canonical mode, so this would require rewriting the CDDL for that use case. This is not prohibitive, just a request.

Reply:
If "text" has been moved up, this is an error that has been introduced accidentally as part of a refactoring to make the specification closer to idiomatic CDDL. There are recipient-necessary fields and recipient unnecessary fields. Text (and, arguably, CoSWID) fall into the recipient-unnecessary category, so they should be pushed to the back. In general, I have a preference for a strict ordering of groups of fields: 

1. unseverable, recipient-necessary
2. severable, recipient-necessary
3. unseverable, recipient-unnecessary
4. severable, recipient-unnecessary

Validate, Load, Run fall into Category 1. Dependency Resolution, Fetch, Install fall into Category 2. I'm not aware of any elements for Category 3. Text and CoSWID fall into Category 4.

That's the reason for the original ordering and that original ordering explains the numbers present in the tags today. If the CDDL has changed to alter that order, then those CDDL changes are, in my opinion, incorrect, and I'd rather fix the CDDL than alter the element order.

Reply:
I see. Changing the CDDL instead sounds good to me.